services/rollout-service/pkg/argo/argo_test.go

```diff
@@ -963,96 +1002,490 @@ func TestCreateOrUpdateArgoApp(t *testing.T) {
+					Event:             "ADDED",
+					ManifestPath:      "/environments/staging/applications/foo/manifests",
+				},
+				{
+					Name:              "foo",
+					Environment:       "test-staging-de-2",
+					ParentEnvironment: "staging",
+					Event:             "ADDED",
+					ManifestPath:      "/environments/staging/applications/foo/manifests",
+				},
+				{
+					Name:              "foo2",
+					Environment:       "test-staging-de-1",
+					ParentEnvironment: "staging",
+					Event:             "ADDED",
+					ManifestPath:      "/environments/staging/applications/foo2/manifests",
+				},
+				{
+					Name:              "foo2",
+					Environment:       "test-staging-de-2",
+					ParentEnvironment: "staging",
+					Event:             "ADDED",
+					ManifestPath:      "/environments/staging/applications/foo2/manifests",
+				},
+				{
+					Name:              "foo",
+					Environment:       "test-staging-de-1",
+					ParentEnvironment: "staging",
+					Event:             "DELETED",
+					ManifestPath:      "/environments/staging/applications/foo/manifests",
+				},
+				{
+					Name:              "foo",
+					Environment:       "test-staging-de-2",
+					ParentEnvironment: "staging",
+					Event:             "DELETED",
+					ManifestPath:      "/environments/staging/applications/foo/manifests",
+				},
+			},
+		},
 	}
+	for _, tc := range tcs {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			as := &mockApplicationServiceClient{
+				Steps:     make([]step, 0),
+				cancel:    cancel,
+				t:         t,
+				lastEvent: make(chan *ArgoEvent, 10),
+			}
+			hlth := &setup.HealthServer{}
+			argoProcessor := &ArgoAppProcessor{
+				lastOverview:          tc.ArgoOverview[0],
+				ApplicationClient:     as,
+				trigger:               make(chan *ArgoOverview, 10),
+				ArgoApps:              make(chan *v1alpha1.ApplicationWatchEvent, 10),
+				ManageArgoAppsEnabled: true,
+				ManageArgoAppsFilter:  []string{"*"},
+				KnownApps:             map[string]map[string]*v1alpha1.Application{},
+			}
+			argoProcessor.PopulateAppsToKnownApps(tc.KnowArgoApps)
+			as.PopulateApps(tc.KnowArgoApps)
+			hlth.BackOffFactory = func() backoff.BackOff { return backoff.NewConstantBackOff(0) }
+			errCh := make(chan error)
+			go func() {
+				errCh <- ConsumeArgo(ctx, hlth.Reporter("consume-argo"), as, argoProcessor.ArgoApps)
+			}()
 
-	for i := range toDelete {
-		a.ApplicationClient.Delete(ctx, &application.ApplicationDeleteRequest{
-			Name: conversion.FromString(toDelete[i].Name),
-		})
+			go func() {
+				errCh <- argoProcessor.Consume(ctx, hlth.Reporter("consume"))
+			}()
 
-	}
-}
+			for _, ov := range tc.ArgoOverview {
+				argoProcessor.Push(ctx, ov)
+			}
 
-func (a *mockArgoProcessor) GetManageArgoAppsFilter() []string {
-	return a.ManageArgoAppsFilter
-}
+			//We add a delay so that all the events are reported by the application client
+			time.Sleep(10 * time.Second)
```

Comment:
can we replace this with a proper wait?
e.g. wait until `errCh` has 2 elements.
It seems like we're not doing anything with `errCh` here, which sounds wrong.